Kuzzle.Tests/API/Controllers/IndexControllerTest.cs

```diff
@@ -0,0 +1,202 @@
+﻿using System;
+using System.Collections.Generic;
+using KuzzleSdk.API.Controllers;
+using KuzzleSdk.API.DataObjects;
+using KuzzleSdk.API.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Kuzzle.Tests.API.Controllers {
+  public class IndexControllerTest {
+    private readonly IndexController _indexController;
+    private readonly KuzzleApiMock _api;
+
+    public IndexControllerTest() {
+      _api = new KuzzleApiMock();
+      _indexController = new IndexController(_api.MockedObject);
+    }
+
+    [Fact]
+    public async void CreateAsyncTest() {
+      _api.SetResult(@"
+        { 
+          result: {
+            acknowledged: true, 
+            shards_acknowledged: true
+          }
+        }
+      ");
+
+      Assert.Equal(
+          new JObject {
+            { "acknowledged", true },
+            { "shards_acknowledged", true }
+          }, await _indexController.CreateAsync("foo")
+      );
+      
+      _api.Verify(new JObject {
+        { "controller", "index" },
+        { "action", "create" },
+        { "index", "foo" }
+      });
+    }
+
+    [Fact]
+    public async void DeleteAsyncTest() {
+      _api.SetResult(@"
+        {
+          result: {
+            acknowledged: true
+          } 
+        }
+      ");
+
+      await _indexController.DeleteAsync("foo");
+
+      _api.Verify(new JObject {
+        { "controller", "index" },
+        { "action", "delete" },
+        { "index", "foo" }
+      });
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public async void ExistsAsyncTest(bool result) {
+      _api.SetResult($"{{result:{result.ToString().ToLower()}}}");
```

Comment:
It may be a much clearer way of writing 
```suggestion
      _api.SetResult(new JObject {{ "result" , result }} );
```
(Same for `GetAutoRefreshAsyncTest` & `SetAutoRefreshAsyncTest` )